extensions/mentions/js/src/forum/addComposerAutocomplete.js

```diff
@@ -10,25 +10,39 @@ import highlight from 'flarum/common/helpers/highlight';
 import KeyboardNavigatable from 'flarum/forum/utils/KeyboardNavigatable';
 import { truncate } from 'flarum/common/utils/string';
 import { throttle } from 'flarum/common/utils/throttleDebounce';
+import Badge from 'flarum/common/components/Badge';
 
 import AutocompleteDropdown from './fragments/AutocompleteDropdown';
 import getMentionText from './utils/getMentionText';
 
 const throttledSearch = throttle(
   250, // 250ms timeout
-  function (typed, searched, returnedUsers, returnedUserIds, dropdown, buildSuggestions) {
+  function (typed, searched, returnedUsers, returnedUserIds, returnedGroups, returnedGroupIds, dropdown, buildSuggestions) {
     const typedLower = typed.toLowerCase();
     if (!searched.includes(typedLower)) {
-      app.store.find('users', { filter: { q: typed }, page: { limit: 5 } }).then((results) => {
-        results.forEach((u) => {
-          if (!returnedUserIds.has(u.id())) {
-            returnedUserIds.add(u.id());
-            returnedUsers.push(u);
-          }
+      if (app.forum.attribute('canSearchUsers')) {
+        app.store.find('users', { filter: { q: typed }, page: { limit: 5 } }).then((results) => {
+          results.forEach((u) => {
+            if (!returnedUserIds.has(u.id())) {
+              returnedUserIds.add(u.id());
+              returnedUsers.push(u);
+            }
+          });
         });
+      }
 
-        buildSuggestions();
-      });
+      if (app.forum.attribute('canSearchGroups')) {
+        app.store.find('groups', { filter: { q: typed }, page: { limit: 5 } }).then((results) => {
+          results.forEach((g) => {
+            if (!returnedGroupIds.has(g.id())) {
+              returnedGroupIds.add(g.id());
+              returnedGroups.push(g);
+            }
+          });
+        });
+      }
+
+      buildSuggestions();
```

Comment:
I believe `buildSuggestions` should be called once we're certain both API requests have finished, like before